# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerService.h
# Commit: df07abc82381
# Full Hash: df07abc82381d2a3067b7de63867a0181f497519
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-26 09:13:23
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P2 PRemoteWorkerDebuggerMangaer binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebuggerManager binding.
#   
#   PRemoteWorkerDebuggerManager is the IPC that handles the registration/unregistration of remote debuggers.
# ==============================================================================

diff -r 5d39af040a61 -r df07abc82381 dom/workers/remoteworkers/RemoteWorkerService.h
--- a/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Mar 25 14:33:27 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Mar 25 14:33:27 2025 +0000
@@ -18,9 +18,12 @@
 
 namespace mozilla::dom {
 
+class RemoteWorkerDebuggerManagerChild;
 class RemoteWorkerService;
 class RemoteWorkerServiceChild;
 class RemoteWorkerServiceShutdownBlocker;
+class PRemoteWorkerDebuggerManagerChild;
+class PRemoteWorkerDebuggerParent;
 class PRemoteWorkerServiceChild;
 
 /**
@@ -71,9 +74,14 @@
   // To be called when a process is initialized on main-thread.
   static void InitializeParent();
   static void InitializeChild(
-      mozilla::ipc::Endpoint<PRemoteWorkerServiceChild> aEndpoint);
+      mozilla::ipc::Endpoint<PRemoteWorkerServiceChild> aEndpoint,
+      mozilla::ipc::Endpoint<PRemoteWorkerDebuggerManagerChild>
+          aDebuggerChildEp);
 
   static nsIThread* Thread();
+  static void RegisterRemoteDebugger(
+      RemoteWorkerDebuggerInfo aDebuggerInfo,
+      mozilla::ipc::Endpoint<PRemoteWorkerDebuggerParent> aDebuggerParentEp);
 
   // Called by RemoteWorkerChild instances on the "Worker Launcher" thread at
   // their creation to assist in tracking when it's safe to shutdown the
@@ -96,7 +104,9 @@
   ~RemoteWorkerService();
 
   nsresult InitializeOnMainThread(
-      mozilla::ipc::Endpoint<PRemoteWorkerServiceChild> aEndpoint);
+      mozilla::ipc::Endpoint<PRemoteWorkerServiceChild> aEndpoint,
+      mozilla::ipc::Endpoint<PRemoteWorkerDebuggerManagerChild>
+          aDebuggerChildEp);
 
   void InitializeOnTargetThread(
       mozilla::ipc::Endpoint<PRemoteWorkerServiceChild> aEndpoint);
@@ -113,6 +123,7 @@
 
   nsCOMPtr<nsIThread> mThread;
   RefPtr<RemoteWorkerServiceChild> mActor;
+  RefPtr<RemoteWorkerDebuggerManagerChild> mDebuggerManagerActor;
   // The keep-alive is set and cleared on the main thread but we will hand out
   // additional references to it from the "Worker Launcher" thread, so it's
   // appropriate to use a mutex.  (Alternately we could have used a ThreadBound
